text/0029-hammer-help-formatting.md

```diff
@@ -0,0 +1,260 @@
+It might make sense to remove such descriptions.
+
+
+### 4. non-uniform capitalization in the descriptions
+
+We have different capitalization in option descriptions accross the whole hammer. See following part of `hammer repository create -h`:
+```
+#...
+--upstream-password UPSTREAM_PASSWORD       Password of the upstream repository user used for authentication
+--upstream-username UPSTREAM_USERNAME       Username of the upstream repository user used for authentication
+--url URL                                   repository source url
+--verify-ssl-on-sync VERIFY_SSL_ON_SYNC     if true, Katello will verify the upstream url’s SSL certifcates are signed by a trusted CA.
+                                            One of true/false, yes/no, 1/0.
+-h, --help                                  print help
+```
+
+# Proposed solutions
+[solution]: #proposed-solutions
+
+### 1. long option lists are difficult to read
+
+Resource identifier options can be squeezed to one line.
+
+```
+before:
+--architecture ARCHITECTURE_NAME       Architecture name
+--architecture-id ARCHITECTURE_ID
+
+after:
+--architecture[-id] ARCHITECTURE       Architecture
+```
+
+```
+before:
+--organization ORGANIZATION_NAME         Organization name to search by
+--organization-id ORGANIZATION_ID        organization ID
+--organization-label ORGANIZATION_LABEL  Organization label to search by
+
+after:
+ --organization[-id|-label] ORGANIZATION  Organization
+```
+
+We can discuss whether it makes sense to keep the simple parameter description there ("Architecture", "Organization").
+
+Rough stats of how squeezing the resource identifier options helps to make the output shorter for a few commands:
+
+| Command           | Option lines before | Option lines after |
+| ----------------- | --- | --- |
+| host create       | 82  | 60  |
+| cvv list          | 22  | 17  |
+| repository create | 25  | 21  |
+| repository list   | 27  | 20  |
+
+
+### 2. no information about required options
+
+Shortening the options from 1. would make it easier to mark a group of resource identifiers as required
+where it wasn't possible before due to their mutual exclusivity:
+```
+ --organization[-id|label] ORGANIZATION     (required)
+```
+
+Option dependencies can be reflected a) the same way on the option line:
+```
+--content-view[-id] CONTENT_VIEW        Content view of the content view version (required for --content-view-version)
+--organization[-id|label] ORGANIZATION  (required for --product and --content-view)
+```
+
+b) described in a free text below the option list:
+```
+--product requires --organization[-id]
+--content-view requires --organization[-id]
+--content-view-version requires --content-view[-id]
+```
+
+or c) below the option list as a more condensed text:
+```
+--product and --content-view requires --organization[-id]
+--content-view-version requires --content-view[-id]
+```
+
+The dependency descriptions can be generated from current apidocs.
```

Comment:
I prefer b) since I'd like to see what is also required when I use this specific option. a) + b) altogether would be nice